zerver/lib/actions.py

```diff
@@ -2666,14 +2666,14 @@ def validate_user_access_to_subscribers_helper(
     if stream_dict["is_web_public"]:
         return
 
-    # Guest users can access subscribed public stream's subscribers
+    # With the exception of web public streams, a guest must
+    # be subscribed to a stream (even a public one) in order
+    # to see subscribers.
     if user_profile.is_guest:
         if check_user_subscribed(user_profile):
             return
-        # We could put an AssertionError here; in that we don't have
-        # any code paths that would allow a guest user to access other
-        # streams in the first place.
-
+        else:
+            raise JsonableError(_("Subscriber data is not available for this stream"))
```

Comment:
Do we need this line?  I think falling through, the `user_profile.can_access_public_streams` is equivalent to checking if the user is a guest, right?

Reply:
I think explicit is better here, but I added a commit instead that just makes the comment more accurate about the current code structure.  (I haven't pushed yet, still dealing with other comments.)